src/clu_main.c

```diff
@@ -0,0 +1,185 @@
+                break;
+            }
+            current = mode_options[i];
+            i = i+1;
+        }
+    }
+    return ret;
+}
+
+
+
+int main(int argc, char** argv)
+{
+    int     flag = 0;
+    int     ret = 0;
+    int     longIndex = 0;
+
+    if (argc == 1) {
+        printf("Main Help.\n");
+        wolfCLU_help();
+    }
+
+    /* If the first string does not have a '-' in front of it then try to
+     * get the mode to use i.e. x509, req, version ... this is for
+     * compatibility with the behavior of the OpenSSL command line utility
+     */
+    if (argc > 1 && argv[1] != NULL && argv[1][0] != '-') {
+        flag = getMode(argv[1]);
+    }
+    else {
+        /* retain old version of modes where '-' is used. i.e -x509, -req */
+        flag = getopt_long_only(argc, argv,"", mode_options, &longIndex);
+    }
+
+    switch (flag) {
+        case 0:
+            printf("No mode provided.\n");
+            ret = 0;
+            break;
+
+        case WOLFCLU_CRYPT:
+            /* generic 'enc' used, default to encrypt unless -d was used */
+            ret = wolfCLU_checkForArg("-d", 2, argc, argv);
+            if (ret > 0) {
+                ret = wolfCLU_setup(argc, argv, 'd');
+            }
+            else {
+                ret = wolfCLU_setup(argc, argv, 'e');
+            }
+            break;
+
+        case WOLFCLU_ENCRYPT:
+            ret = wolfCLU_setup(argc, argv, 'e');
+            break;
+
+        case WOLFCLU_DECRYPT:
+            ret = wolfCLU_setup(argc, argv, 'd');
+            break;
+
+        case WOLFCLU_BENCHMARK:
+            ret = wolfCLU_benchSetup(argc, argv);
+            break;
+
+        case WOLFCLU_HASH:
+            ret = wolfCLU_hashSetup(argc, argv);
+            break;
+
+        case WOLFCLU_MD5:
+            ret = wolfCLU_md5Setup(argc, argv);
+            break;
+
+        case WOLFCLU_X509:
+            ret = wolfCLU_certSetup(argc, argv);
+            break;
+
+        case WOLFCLU_REQUEST:
+            ret = wolfCLU_requestSetup(argc, argv);
+            break;
+
+        case WOLFCLU_GEN_KEY:
+            ret = wolfCLU_genKeySetup(argc, argv);
+            break;
+
+        case WOLFCLU_ECPARAM:
+            ret = wolfCLU_ecparam(argc, argv);
+            break;
+
+        case WOLFCLU_PKEY:
+            ret = wolfCLU_pKeySetup(argc, argv);
+            break;
+
+        case WOLFCLU_DGST:
+            ret = wolfCLU_dgst_setup(argc, argv);
+            break;
+
+        case WOLFCLU_RSA:
+        case WOLFCLU_ECC:
+        case WOLFCLU_ED25519:
+            ret = wolfCLU_sign_verify_setup(argc, argv);
+            break;
+
+        case WOLFCLU_HELP:
+            /* only print for -help if no mode has been declared */
+            printf("Main help menu:\n");
+            wolfCLU_help();
+            return 0;
+
+        case WOLFCLU_VERBOSE:
+            wolfCLU_verboseHelp();
+            return 0;
+
+        case 'v':
+            (void)wolfCLU_version();
```

Comment:
Because wolfCLU_version returns an int value some static analyis tools will complain about the return value not being checked. Since we are not doing anything with the return value anyway, casting as void will silence the warnings.